centreon/src/Centreon/Infrastructure/Contact/ContactRepositoryRDB.php

```diff
@@ -55,14 +55,17 @@ public function findById(int $contactId): ?Contact
     {
         $request = $this->translateDbName(
             'SELECT contact.*, cp.password AS contact_passwd, t.topology_url,
-            t.topology_url_opt, t.is_react, t.topology_id, tz.timezone_name
+            t.topology_url_opt, t.is_react, t.topology_id, tz.timezone_name,
+            COALESCE(contact.default_page, template.default_page) as default_page
             FROM `:db`.contact
+            LEFT JOIN centreon.contact as template
+                ON contact.contact_template_id = template.contact_id
             LEFT JOIN `:db`.contact_password cp
                 ON cp.contact_id = contact.contact_id
             LEFT JOIN `:db`.timezone tz
                 ON tz.timezone_id = contact.contact_location
             LEFT JOIN `:db`.topology t
-                ON t.topology_page = contact.default_page
+                ON t.topology_page = COALESCE(contact.default_page, template.default_page)
```

Comment:
If the **LEFT JOIN** is done over `COALESCE(contact.default_page, template.default_page)`, wouldn't be logical to use `t.topology_page` in the **SELECT** instead of `COALESCE(contact.default_page, template.default_page)` ?